MobileSyncExplorerKotlinTemplate/app/src/main/java/com/salesforce/mobilesyncexplorerkotlintemplate/core/extensions/NumberExt.kt

```diff
@@ -0,0 +1,3 @@
+package com.salesforce.mobilesyncexplorerkotlintemplate.core.extensions
+
+fun UInt.coerceToNonNegativeInt(): Int = coerceIn(0u..Int.MAX_VALUE.toUInt()).toInt()
```

Comment:
Ever since unsigned types were stable I've been trying to use them for enforcing non-negative values in methods, but so many APIs still need `int` that it makes it cumbersome to use.  I'll remove this extension and just do the conversion in-line.